nltk/text.py

```diff
@@ -530,16 +534,50 @@ def dispersion_plot(self, words):
         from nltk.draw import dispersion_plot
 
         dispersion_plot(self, words)
-
-    def generate(self, words):
-        """
-        Issues a reminder to users following the book online
-        """
-        import warnings
-
-        warnings.warn(
-            'The generate() method is no longer available.', DeprecationWarning
-        )
+        
+    def build_ngram_lm(self, tokenized_sents, n=3):
+        train_data, padded_sents = padded_everygram_pipeline(n, tokenized_sents)
+        model = MLE(order=n)
+        model.fit(train_data, padded_sents)
+        return model
+        
+    def generate(self, length=100, text_seed=None, random_seed=42):
+        """
+        Print random text, generated using a trigram language model.
+        See also `help(nltk.lm)`.
+        
+        :param length: The length of text to generate (default=100)
+        :type length: int
+        
+        :param text_seed: Generation can be conditioned on preceding context.
+        :type text_seed: list(str)
+        
+        :param random_seed: A random seed or an instance of `random.Random`. If provided,
+        makes the random sampling part of generation reproducible.
+        :type random_seed: int
+        
+        """
+        # Create the model when using it the first time.
+        self.tokenized_sents = [sent.split(' ') for sent in 
+                                sent_tokenize(' '.join(self.tokens))]
+        if not hasattr(self, 'trigram_model'):
+            print("Building ngram index...", file=sys.stderr)
+            self.trigram_model = self.build_ngram_lm(self.tokenized_sents, n=3)
+            
+        generated_tokens = []
+        
+        assert length > 0, 'The `length` must be more than 0.'
+        while len(generated_tokens) < length:
+            for idx, token in enumerate(self.trigram_model.generate(length, text_seed=text_seed, random_seed=random_seed)):
+                if token == '<s>':
+                    continue
+                if token == '</s>':
+                    break
+                generated_tokens.append(token)
+            random_seed += 1
+        
+        prefix = ' '.join(text_seed) + ' ' if text_seed else ''
+        print(prefix + tokenwrap(generated_tokens[:length]))
```

Comment:
I'm surprised to see `print` and not `return` here. It really seems like this is intended to be kind of a "display" or "demo" method and its output not passed on to other functions. Right?

Reply:
Yeah, the original Text was presented as a demo. But I agree, maybe doing both would be good, print as well as return.

Reply:
yea, let's do that for posterity

Reply:
Done.